File: backend/letscookLibrary/letscookLibrary/Recipe.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace letscookLibrary
{
    public class Recipe: BaseEntity
    {
        public string Name { get; set; }
        public int NumberOfPeople { get; set; }
        public int PreparationTime { get; set; }
        public int CookingTime { get; set; }
        public bool Favorite { get; set; }
        public string PhotoPath{ get; set; }
        public List<Ingredient> Ingredients { get; set; }
        public List<string> Directions { get; set; }
        public List<string> Hashtags { get; set; }
    }
}

File: frontend/nativescript/src/app/model/recipe.ts
import { Ingredient } from "./ingredient";
import { BaseEntity } from "./base-entity";

export class Recipe extends BaseEntity {
    name: string;
    numberOfPeople: number = 0;
    preparationTime: number = 0;
    cookingTime: number = 0;
    favorite: boolean = false;
    photoPath: string = undefined;
    ingredients: Array<Ingredient> = [];
    directions: Array<string> = [];
    hashtags: Array<string> = [];
}
